vault/data_source_aws_static_credentials.go

```diff
@@ -0,0 +1,79 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+const staticCredsAffix = "static-creds"
+
+func awsStaticCredDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: ReadContextWrapper(awsStaticCredentialsDataSourceRead),
+		Schema: map[string]*schema.Schema{
+			consts.FieldMount: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "AWS Secret Backend to read credentials from.",
+			},
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role.",
+			},
+			consts.FieldAccessKey: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "AWS access key ID read from Vault.",
+				Sensitive:   true,
+			},
+			consts.FieldSecretKey: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "AWS secret key read from Vault.",
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func awsStaticCredentialsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mount := d.Get(consts.FieldMount).(string)
+	role := d.Get(consts.FieldName).(string)
+	fullPath := fmt.Sprintf("%s/%s/%s", mount, staticCredsAffix, role)
+
+	secret, err := client.Logical().ReadWithContext(ctx, fullPath)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading from Vault: %s", err))
+	}
+	log.Printf("[DEBUG] Read %q from Vault", fullPath)
+	if secret == nil {
+		return diag.FromErr(fmt.Errorf("no role found at %q", fullPath))
+	}
+
+	d.SetId(fullPath)
+
+	// Unfortunately we called the secret fields in vault different things between the dynamic and static credentials;
+	// so we're left with the choice here of keeping them common among terraform data sources, or keeping them
+	// the same between vault and terraform.
```

Comment:
Out of scope for this PR, but is it worthwhile changing the naming for static creds in Vault? Would that improve things here?

Reply:
it could - i'm not entirely sure what our style guidance is here regarding what should match (i suppose everything everywhere _should_ match, ideally)

Reply:
I think at the very least we should try make sure the naming within Vault is consistent between dynamic and static credentials. If that isn't possible then we should do best effort to make it as straightforward as possible.

Reply:
Yeah I think Vault is very inconsistent about the access key fields for AWS. I have seen `access_key`, `access_key_id` and `aws_access_key_id` 😓 

I think it's best to match the API field in Vault for less confusion. We have precedents for that with other AWS resources. So I would suggest making the schema field `FieldAccessKeyID` (you might have to add a constant for that)

Reply:
Updating this, thanks to Calvin and JM we got the field name changed merged into vault.